src/providers/tpm_provider/utils.rs

```diff
@@ -243,6 +243,72 @@ pub fn parsec_to_tpm_signature(
     })
 }
 
+/// Validates an RSAPublicKey against the attributes we expect. Returns ok on success, otherwise
+/// returns an error.
+pub fn validate_public_key(public_key: &RSAPublicKey, attributes: &Attributes) -> Result<()> {
+    if public_key.modulus.is_negative() || public_key.public_exponent.is_negative() {
+        error!("Only positive modulus and public exponent are supported.");
+        return Err(ResponseStatus::PsaErrorInvalidArgument);
+    }
+
+    if public_key.public_exponent.as_unsigned_bytes_be() != PUBLIC_EXPONENT {
+        if crate::utils::GlobalConfig::log_error_details() {
+            error!("The TPM Provider only supports 0x101 as public exponent for RSA public keys, {:?} given.", public_key.public_exponent.as_unsigned_bytes_be());
+        } else {
+            error!("The TPM Provider only supports 0x101 as public exponent for RSA public keys");
+        }
+        return Err(ResponseStatus::PsaErrorNotSupported);
+    }
+    let key_data = public_key.modulus.as_unsigned_bytes_be();
+    let len = key_data.len();
+
+    let key_bits = attributes.bits;
+    if key_bits != 0 && len * 8 != key_bits {
+        if crate::utils::GlobalConfig::log_error_details() {
+            error!(
+                    "`bits` field of key attributes (value: {}) must be either 0 or equal to the size of the key in `data` (value: {}).",
+                    attributes.bits,
+                    len * 8
+                );
+        } else {
+            error!("`bits` field of key attributes must be either 0 or equal to the size of the key in `data`.");
+        }
+        return Err(ResponseStatus::PsaErrorInvalidArgument);
+    }
+
+    if len != 128 && len != 256 {
+        if crate::utils::GlobalConfig::log_error_details() {
+            error!(
+                "The TPM provider only supports 1024 and 2048 bits RSA public keys ({} bits given).",
```

Comment:
Is this limitation in the `TransientKeyContext`? I think TPMs support one of `1024, 2048, 3072 or 4096`